test/foundry/Certify.t.sol

```diff
@@ -0,0 +1,371 @@
+
+        // Call deposit to make sure system is certified
+        vm.expectEmit(false, false, false, true);
+        emit DepositWithReceipt(bob, alice, assets, assets, 1, data);
+        vault.deposit(assets, alice, minShareRatio, data);
+
+        vm.stopPrank();
+    }
+
+    /// Test certify with force until true in future
+    function testCertifyWithForceUntilTrueInFuture(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        uint256 assets,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 certifyUntilFuture,
+        uint256 certifyUntil,
+        uint256 referenceBlockNumber,
+        bytes memory data,
+        uint256 blockNumber,
+        uint256 minShareRatio
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address bob = vm.addr((fuzzedKeyBob % (SECP256K1_ORDER - 1)) + 1);
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+
+        // Assume that assets is less uint256 max
+        assets = bound(assets, 1, type(uint256).max);
+
+        // Ensure blockNumber is within the valid range
+        blockNumber = bound(block.number, 0, type(uint256).max);
+        vm.roll(blockNumber);
+        referenceBlockNumber = bound(referenceBlockNumber, 0, blockNumber);
+        certifyUntilFuture = bound(certifyUntilFuture, blockNumber + 1, type(uint32).max);
+        certifyUntil = bound(certifyUntil, 1, type(uint32).max);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Prank as Alice to grant role
+        vm.startPrank(alice);
+
+        // Grant CERTIFIER role to Alice
+        vault.grantRole(vault.CERTIFIER(), bob);
+        vault.grantRole(vault.DEPOSITOR(), bob);
+
+        // Prank Bob for the transaction
+        vm.startPrank(bob);
+
+        // Non-Force certify to be overwritten later
+        vm.expectEmit(false, false, false, true);
+        emit Certify(bob, certifyUntil, referenceBlockNumber, false, data);
+        vault.certify(certifyUntil, referenceBlockNumber, false, data);
+
+        // Force certify in the future
+        vm.expectEmit(false, false, false, true);
+        emit Certify(bob, certifyUntilFuture, referenceBlockNumber, true, data);
+        vault.certify(certifyUntilFuture, referenceBlockNumber, true, data);
+
+        // Set up the event expectation for DepositWithReceipt
+        vm.expectEmit(false, false, false, true);
+        emit DepositWithReceipt(bob, alice, assets, assets, 1, data);
+        vault.deposit(assets, alice, minShareRatio, data);
+
+        vm.stopPrank();
+    }
+
+    /// Test certify with force until true in present
+    function testCertifyWithForceUntilTrueInPresent(
```

Comment:
i don't want three separate past/present/future tests... i want one fuzzed test that just shows force working for all values